json/src/main/scala/com/gu/contentapi/json/Serialization.scala

```diff
@@ -69,8 +71,25 @@ object Serialization {
     case JField("safeEmbedCode", JString(s)) => JField("safeEmbedCode", JBool(s.toBoolean))
   }
 
-  def thriftEnum2JString[A <: ThriftEnum]: PartialFunction[Any, JString] = {
-    case a: ThriftEnum => JString(pascalCaseToHyphenated(a.name))
+  def thriftEnum2JString[A <: ThriftEnum](implicit classTag: ClassTag[A]): PartialFunction[Any, JString] = {
+    case a if classTag.runtimeClass.isInstance(a) => JString(pascalCaseToHyphenated(a.asInstanceOf[A].name))
+  }
```

Comment:
Recap of what I explained offline:

* `case a: ThriftEnum` means the serializer will be used for all Thrift enums, not just those of type `A`. This is quite surprising behaviour of json4s, and it means that we can't customise the serialization of particular Thrift enums.
* We really want to write `case a: A`, but that doesn't work because the type `A` is erased at runtime.
* We work around it using `ClassTag`, which preserves compile-time class information to make it available at runtime.
* `ClassTag` is always available in implicit scope. The compiler supplies it for us.